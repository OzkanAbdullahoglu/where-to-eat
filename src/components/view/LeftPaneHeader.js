/* eslint-disable linebreak-style */
import React from 'react';
import PropTypes from 'prop-types';


const LeftPaneHeader = ({
  userLocationUpdate,
  setTogglePaneOnly,
}) => (
  <div>
    <div
      className="cards-layout cards-layout-flex-horizontal"
      tabIndex="-1"
    >
      <button className="header-buttons" onClick={userLocationUpdate}>
                    Use my location
      </button>
      <button
        className="header-buttons hide-larger-device"
        onClick={setTogglePaneOnly}
      >
                    Show on map
      </button>
    </div>
    <div className="cards-divider cards-divider-bottom-line"></div>
  </div>
);

LeftPaneHeader.propTypes = {
  userLocationUpdate: PropTypes.func,
  setTogglePaneOnly: PropTypes.func,
};

export default LeftPaneHeader;
