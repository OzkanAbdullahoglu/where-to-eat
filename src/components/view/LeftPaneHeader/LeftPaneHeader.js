/* eslint-disable linebreak-style */
import React from 'react';
import PropTypes from 'prop-types';

const LeftPaneHeader = ({
  userLocationUpdate,
  setTogglePaneOnly,
}) => (
  <div>
    <div
      className="cards-layout cards-layout-flex-horizontal"
      tabIndex="-1"
    >
      <button className="use-my-location header-buttons" onClick={userLocationUpdate}>
                    Use my location
      </button>
      <button
        className="set-toggle-pane-only header-buttons hide-larger-device"
        onClick={setTogglePaneOnly}
      >
                    Show on map
      </button>
    </div>
    <div className="cards-divider cards-divider-bottom-line"></div>
  </div>
);

LeftPaneHeader.propTypes = {
  userLocationUpdate: PropTypes.func,
  setTogglePaneOnly: PropTypes.func,
};

export default LeftPaneHeader;
