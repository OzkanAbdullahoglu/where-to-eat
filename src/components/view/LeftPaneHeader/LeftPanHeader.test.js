/* eslint-disable linebreak-style */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import LeftPaneHeader from './LeftPaneHeader';

describe('Left Panel Header Component', () => {
  const userLocationUpdateMocked = jest.fn();
  const setTogglePaneOnlyMocked = jest.fn();
  const props = {
    userLocationUpdate: userLocationUpdateMocked,
    setTogglePaneOnly: setTogglePaneOnlyMocked,
  };
  const leftPaneHeader = shallow(<LeftPaneHeader {...props} />);
  it('renders Left Panel Header Component', () => {
    expect(toJson(leftPaneHeader)).toMatchSnapshot();
  });
  it('calls `userLocationUpdate` prop function once', () => {
    const useMyLocationButton = leftPaneHeader.find('.use-my-location');
    useMyLocationButton.simulate('click');
    const userLocationUpdateCallCount = userLocationUpdateMocked.mock.calls.length;
    expect(userLocationUpdateCallCount).toBe(1);
  });
  it('calls `setTogglePaneOnly` prop function once', () => {
    const setTogglePaneOnlyButton = leftPaneHeader.find('.set-toggle-pane-only');
    setTogglePaneOnlyButton.simulate('click');
    const setTogglePaneOnlyCallCount = setTogglePaneOnlyMocked.mock.calls.length;
    expect(setTogglePaneOnlyCallCount).toBe(1);
  });
});
