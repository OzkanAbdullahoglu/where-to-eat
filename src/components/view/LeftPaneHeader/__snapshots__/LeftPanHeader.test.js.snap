// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Left Panel Header Component renders Left Panel Header Component 1`] = `
<div>
  <div
    className="cards-layout cards-layout-flex-horizontal"
    tabIndex="-1"
  >
    <button
      className="use-my-location header-buttons"
      onClick={[MockFunction]}
    >
      Use my location
    </button>
    <button
      className="set-toggle-pane-only header-buttons hide-larger-device"
      onClick={[MockFunction]}
    >
      Show on map
    </button>
  </div>
  <div
    className="cards-divider cards-divider-bottom-line"
  />
</div>
`;
